src/s3/S3Controller.ts

```diff
@@ -172,11 +146,13 @@ export class S3Controller {
     };
   }
 
-  public isValidLink(link: string, requiredStrings: string[]) {
-    // TODO-kev: Keep as regex or use this hack?
-    return (
-      link.includes(this.publicFileLinkPrefix) &&
-      requiredStrings.every((string) => link.includes(string))
-    );
+  public isValidLink(link: string, subsequentString: string) {
+    // Match format: <publicFileLinkPrefix><subsequentString>
+    const patternString = `^${this.publicFileLinkPrefix}${subsequentString}.*`;
+    const pattern = new RegExp(patternString);
+    const match = link.match(pattern);
+
+    // TODO-kev: or change to !!match?
+    return Boolean(match);
```

Comment:
Let's:
* Get rid of this isValidLink function so we don't expose publicFileLinkPrefix outside of the controller
* Do the pending / approved check part of the link in the agent
* Check that every link passed into the controller has the publicFileLinkPrefix in it